### ISSUE TITLE
chisel printf is unordered

### ISSUE THREAD

**Issue by smirolo** [ucb-bar/chisel#20]
Entered as an issue on behalf of Chris:

So i'm trying out the Chisel printf, which is frankly a must-have as I go forward with debugging a parameterized core in Verilog (where I have no printf code written up yet in the Verilog testharness)...

I have code like this in my Chisel:

// print out ROB
for (i <- 0 until num_rob_entries)
{
  printf("  rob[%d] %d %d (PC:0x%x) 0x%x\n"
        , UFix(i)
        , rob.io.debug.entry(i).valid
        , rob.io.debug.entry(i).busy
        , rob.io.debug.entry(i).uop.pc
        , rob.io.debug.entry(i).uop.inst
        )
}

Unfortunately, the order is all jumbled (notice the numbers in the brackets):

 rob[5] 0 0 (PC:0xe0cc2cc10d53011a) 0x19ba8133
 rob[2] 0 0 (PC:0xad9afe5b3d25cd7a) 0x5d437fba
 rob[7] 0 0 (PC:0x1dee813b0abfc715) 0x6f15c9de
 rob[3] 0 0 (PC:0x99489c3289202a12) 0x522a21a4
 rob[14] 0 0 (PC:0x213d974256d1f3ad) 0xd663a4ac
 rob[4] 0 0 (PC:0xc7bec28f2ec5515d) 0x835e7206
 rob[6] 0 0 (PC:0x3672316cc78d408f) 0x13c85527
 etc….

It is also unordered relative to any printfs/$displays in the test harness itself.

In short, I can't make use of Chisel printfs (outside of perhaps catching on-off error conditions, but I should just use Assert for that anyways).
